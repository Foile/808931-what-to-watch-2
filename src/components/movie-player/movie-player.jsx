import React from "react";
import {compose} from "recompose";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import VideoPlayer from "@components/video-player/video-player";
import VideoPlayerProgress from "@components/video-player/video-player-progress/video-player-progress";
import withPlayer from "@hocs/with-player/with-player";
import {MoviePlayerConstants} from "@constants";

const MoviePlayer = ({movie, videoRef, onPlayPause, onClosePlayer, onFullScreen, isPlaying}) => {
  const {name, videoLink, previewImage} = movie;
  return (
    <div className="player">
      <VideoPlayer
        videoRef={videoRef}
        classes="player__video"
        link={videoLink}
        poster={previewImage}
        width={MoviePlayerConstants.WIDTH}
        height={MoviePlayerConstants.HEIGHT}
      />

      <button
        type="button"
        className="player__exit"s
        onClick={onClosePlayer}
      >
        Exit</button>

      <div className="player__controls">
        <VideoPlayerProgress
          videoRef={videoRef}
        />

        <div className="player__controls-row">
          <button
            className="btn player__play"
            type="button"
            onClick = {onPlayPause}
          >
            <svg viewBox="0 0 19 19" width="19" height="19">
              <use xlinkHref={isPlaying ? `#pause` : `#play-s`}></use>
            </svg>
          </button>
          <div className="player__name">{name}</div>
          <button
            className="btn player__full-screen"
            type="button"
            onClick = {onFullScreen}
          >
            <svg viewBox="0 0 27 27" width="27" height="27">
              <use xlinkHref="#full-screen"></use>
            </svg>
          </button>
        </div>
      </div>
    </div>
  );
};

MoviePlayer.defaultProps = {
  movie: {
    name: ``,
    videoLink: ``,
    previewImage: ``,
    runTime: 0,
  },
  videoRef: {},
  onPlayPause: () => {},
  onClosePlayer: () => {},
  onFullScreen: () => {},
  isPlaying: false,
};

MoviePlayer.propTypes = {
  movie: PropTypes.shape({
    name: PropTypes.string.isRequired,
    videoLink: PropTypes.string.isRequired,
    previewImage: PropTypes.string.isRequired,
    runTime: PropTypes.number.isRequired,
  }).isRequired,
  videoRef: PropTypes.shape({
    current: PropTypes.instanceOf(Element)
  }).isRequired,
  onPlayPause: PropTypes.func.isRequired,
  onClosePlayer: PropTypes.func.isRequired,
  onFullScreen: PropTypes.func.isRequired,
  isPlaying: PropTypes.bool.isRequired,
};

const mapStateToProps = (state, ownProps) => ({
  movie: state.data.allFilms && state.data.allFilms.find(({id}) => id === Number(ownProps.match.params.id))
});

export {MoviePlayer};

export default compose(
    connect(mapStateToProps),
    withPlayer
)(MoviePlayer);
