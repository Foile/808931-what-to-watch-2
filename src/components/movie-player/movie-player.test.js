import React from 'react';
import renderer from 'react-test-renderer';
import {MoviePlayer} from './movie-player';

const props = {
  currentMovie: {
    name: `Aviator`,
    videoLink: `http://video.com/video.mp4`,
    previewImage: `http://images.com/image1.png`,
    runTime: 100,
  },
  videoRef: React.createRef(),
  onPlayPause: () => {},
  onClosePlayer: () => {},
  onFullScreen: () => {},
  isPlaying: false,
};

it(`Movie player correctly renders after relaunch`, () => {
  const tree = renderer.create(<MoviePlayer {...props}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
