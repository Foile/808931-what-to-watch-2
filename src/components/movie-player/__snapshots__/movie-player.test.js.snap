// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie player correctly renders after relaunch 1`] = `
<div
  className="player"
>
  <video
    className="player__video"
    controls={false}
    height="100%"
    muted={false}
    poster=""
    width="100%"
  >
    <source
      src=""
      type="video/mp4"
    />
  </video>
  <button
    className="player__exit"
    onClick={[Function]}
    s={true}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          value={0}
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "0%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        00:00:00
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="btn player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
      </button>
      <div
        className="player__name"
      >
        
      </div>
      <button
        className="btn player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
